rclcpp/include/rclcpp/parameter_client.hpp

```diff
@@ -149,13 +143,55 @@ class AsyncParametersClient
   ))
   {
     return rclcpp::create_subscription<rcl_interfaces::msg::ParameterEvent>(
-      node,
+      node_topics_interface,
```

Comment:
Consider not deleting `NodeT && node` as a template parameter and using here:
```cpp
get_node_topics_interface(node),
```

Also, delete the two other templates you added.
[get_node_topics_interface](https://github.com/ros2/rclcpp/blob/master/rclcpp/include/rclcpp/node_interfaces/get_node_topics_interface.hpp) should do the magic.